src/main/java/example/docvalues/DocValuesSearchExample.java

```diff
@@ -0,0 +1,127 @@
+package example.docvalues;
+
+import example.basic.SimpleSearch;
+import org.apache.lucene.document.Field;
+import org.apache.lucene.document.IntPoint;
+import org.apache.lucene.document.SortedNumericDocValuesField;
+import org.apache.lucene.document.StringField;
+import org.apache.lucene.index.*;
+import org.apache.lucene.search.*;
+import org.apache.lucene.search.SortField.Type;
+import org.apache.lucene.store.Directory;
+import org.apache.lucene.store.FSDirectory;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DocValuesSearchExample {
+
+    // ## What are doc values?
+    // 
+    // Doc values are Apache Lucene's column-stride field value storage, that could be used to store, per document: 
+    // - numerics (single- or multi-valued)
+    // - sorted keywords (single or multi-valued) 
+    // - binary data blobs
+    // 
+    // Doc values are the on-disk data structure, built at document index time. They store the same values as the 
+    // _source but in a column-oriented fashion that is way more  efficient for sorting and aggregations and 
+    // are quite fast to access at search time (only the  value for field in question needs to be decoded per hit).
+    // This is in contrast to Lucene's stored document fields,  which store all field values for one document together 
+    // in a row-stride fashion, and are therefore relatively slow to access.
+    // 
+    // [1] https://www.elastic.co/guide/en/elasticsearch/reference/7.10/doc-values.html#doc-values
+    // [2] https://www.elastic.co/blog/sparse-versus-dense-document-values-with-apache-lucene
+    //
+    // ## Creating Documents
+    //
+    // We will create a list of documents, where each document has a single numeric field `order` of type INT.
+    //
+    // We use the `SortedNumericDocValuesField` type to indicate that it's a numeric field with doc values that
+    // will be used for sorting.
+    //
+    private static List<List<IndexableField>> createDocuments() {
+        List<List<IndexableField>> docs = new ArrayList<>();
+        for (int i = 0; i < 30; ++i) {
+            List<IndexableField> doc = new ArrayList<>();
+            doc.add(new StringField("id", Integer.toString(i), Field.Store.YES));
+            doc.add(new IntPoint("order", i));
```

Comment:
@msfroh since this is a basic examples I didn't include multi values, but certainly could do that, wondering to hear your opinion.

Reply:
Yeah -- I think keeping it simple for now makes sense :+1: